### DISCUSSION TITLE
Using Nostr identities for Provenance

### DISCUSSION THREAD

**Discussion by rot13maxi** [ordinals/ord#1460] [Ideas]
copied from #783:

 I'm thinking of using Nostr identities for provenance, something like this:

Create a Nostr key that represents the owner of the collection.
Create and mint an inscription to announce a collection:

{
"name": "Astral Babes",
"maxSupply": 10000,
"owner": npub12344(nostr-pub of owner)
}
Each time a mint occurs, a separate inscription is also minted that links that item to the collection and adds some metadata:

{
 "inscription_id": "inscription-id-of-the-jpg-inscription",
 "collection_id": "inscription-id-of-the-collection-announcement-inscription",
  "name": "Astral Babe 12",
  "traits": ["hot", "sexy"],
 "signature": "signed-by-the-nostr-key-of-collection-owner"
}
The valid 10,000 items in the collection are the first inscriptions to be mined and also have a valid attestation inscription mined after it.

Furthermore, additional Attestation Inscriptions can be minted afterwards to update the metadata

{
 "inscription_id": "asdfasdfasdfasdfasdf",
  "name": "Astral Babe 2",
  "traits": ["sexy", "hot"],
  "signature": "signed by collection owner"
}
Also can mint a "Freeze Metadata" Attestation which can freeze all future updates to an item or an entire collection

{
 "inscription_id": "asdfasdfasdfasdfasdf",
  "freeze": true,
  "signature": "signed-by-collection-owner"
}
I think the key here is to decouple the inscriptions from identity. Inscriptions are messages, whereas identities are nostr keys

**Comment by dannydeezy** [comment]
copying this follow-up also 

could also link items to a collection in batches like this, to reduce dust
```
{
 "collection_id": "inscription-id-of-the-collection-announcement-inscription",
  "items":[{
         "inscription_id": "asdfasfasdf"
         "name": "Astral Babe 1",
         "traits": ["hot"]
      }, {
          "inscription_id": "ccasdfasfasdf"
         "name": "Astral Babe 2",
         "traits": ["sexy"]         
      }, {
         "inscription_id": "ffffaaaasdddsd"
         "name": "Astral Babe 3",
         "traits": ["inquisitive"]
      }
  ],
 "signature": "signed-by-the-nostr-key-of-collection-owner"
}
```

**Comment by dannydeezy** [comment]
the one thing this is missing is it doesn't commit to the contents of the collection prior to launching, but i'm ok with that

**Comment by dannydeezy** [comment]
i think this works...

**Comment by dannydeezy** [comment]
eh actually.. Nostr is not necessary. 
My new thoughts are here: https://github.com/casey/ord/discussions/1469
